crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1778,6 +1778,68 @@ impl<'db> Type<'db> {
                         let revealed_ty = binding.first_parameter().unwrap_or(Type::Unknown);
                         CallOutcome::revealed(binding, revealed_ty)
                     }
+                    Some(KnownFunction::StaticAssert) => {
+                        let parameter_ty = binding.first_parameter().unwrap_or(Type::Unknown);
+                        let truthiness = parameter_ty.bool(db);
+
+                        binding.set_return_ty(Type::none(db));
+
+                        if truthiness.is_always_true() {
+                            CallOutcome::callable(binding)
+                        } else {
+                            CallOutcome::StaticAssertionError {
+                                parameter_ty,
+                                truthiness,
+                            }
+                        }
+                    }
```

Comment:
We currently do not emit a (proper) diagnostic for calls like `reveal_type()` or `static_assert()` (this is a pre-existing issue, pre-dating call-signature checking). I'll look into that tomorrow.